boa_engine/src/builtins/generator_function/mod.rs

```diff
@@ -0,0 +1,131 @@
+#[derive(Debug, Clone, Copy)]
+pub struct GeneratorFunction;
+
+impl BuiltIn for GeneratorFunction {
+    const NAME: &'static str = "GeneratorFunction";
+
+    const ATTRIBUTE: Attribute = Attribute::NON_ENUMERABLE.union(Attribute::WRITABLE);
+
+    fn init(context: &mut Context) -> JsValue {
+        let _timer = Profiler::global().start_event(Self::NAME, "init");
+
+        let prototype = &context
+            .standard_objects()
+            .generator_function_object()
+            .prototype;
+        let constructor = &context
+            .standard_objects()
+            .generator_function_object()
+            .constructor;
+
+        constructor.set_prototype(Some(
+            context.standard_objects().function_object().constructor(),
+        ));
+        let property = PropertyDescriptor::builder()
+            .value(1)
+            .writable(false)
+            .enumerable(false)
+            .configurable(true);
+        constructor.borrow_mut().insert("length", property);
+        let property = PropertyDescriptor::builder()
+            .value("GeneratorFunction")
+            .writable(false)
+            .enumerable(false)
+            .configurable(true);
+        constructor.borrow_mut().insert("name", property);
+        let property = PropertyDescriptor::builder()
+            .value(
+                context
+                    .standard_objects()
+                    .generator_function_object()
+                    .prototype(),
+            )
+            .writable(false)
+            .enumerable(false)
+            .configurable(false);
+        constructor.borrow_mut().insert("prototype", property);
+        constructor.borrow_mut().data = ObjectData::function(Function::Native {
+            function: Self::constructor,
+            constructor: true,
+        });
+
+        prototype.set_prototype(Some(
+            context.standard_objects().function_object().prototype(),
+        ));
+        let property = PropertyDescriptor::builder()
+            .value(
+                context
+                    .standard_objects()
+                    .generator_function_object()
+                    .constructor(),
+            )
+            .writable(false)
+            .enumerable(false)
+            .configurable(true);
+        prototype.borrow_mut().insert("constructor", property);
+        let property = PropertyDescriptor::builder()
+            .value(context.standard_objects().generator_object().prototype())
+            .writable(false)
+            .enumerable(false)
+            .configurable(true);
+        prototype.borrow_mut().insert("prototype", property);
+        let property = PropertyDescriptor::builder()
+            .value("GeneratorFunction")
+            .writable(false)
+            .enumerable(false)
+            .configurable(true);
+        prototype
+            .borrow_mut()
+            .insert(WellKnownSymbols::to_string_tag(), property);
```

Comment:
Was there a problem with `ConstructorBuilder`? Or why do we need to manually initialize the constructor an the prototype of generator function objects?

Reply:
Yeah, the generator object hierarchy is pretty complex. I cant recite the problem 100% right now, but modifying `ConstructorBuilder` did not seem efficient, as it is just this object. You can take a look at the spec for the generator object relationships https://tc39.es/ecma262/#figure-2. Maybe you can come up with a better way to do it, but it seemed good enough to get it working :D

Reply:
Yeah, let's cleanup the code later, the priority right now is having generators integrated :)